ts/parts-map/HeatmapSeries.ts

```diff
@@ -303,82 +352,143 @@ seriesType<Highcharts.HeatmapSeries>(
 
             series.generatePoints();
-
             series.points.forEach(function (
                 point: Highcharts.HeatmapPoint
             ): void {
-                var xPad = (options.colsize || 1) / 2,
-                    yPad = (options.rowsize || 1) / 2,
-                    x1 = clamp(
-                        Math.round(
-                            xAxis.len -
-                            (xAxis.translate(
-                                point.x - xPad,
-                                0 as any,
-                                1 as any,
-                                0 as any,
-                                1 as any,
-                                -pointPlacement
-                            ) as any)
-                        ),
-                        -xAxis.len, 2 * xAxis.len
-                    ),
-                    x2 = clamp(
-                        Math.round(
-                            xAxis.len -
-                            (xAxis.translate(point.x + xPad,
-                                0 as any,
-                                1 as any,
-                                0 as any,
-                                1 as any,
-                                -pointPlacement
-                            ) as any)
-                        ),
-                        -xAxis.len, 2 * xAxis.len
-                    ),
-                    y1 = clamp(
-                        Math.round(yAxis.translate(
-                            point.y - yPad,
-                            0 as any,
-                            1 as any,
-                            0 as any,
-                            1 as any
-                        ) as any),
-                        -yAxis.len, 2 * yAxis.len
-                    ),
-                    y2 = clamp(
-                        Math.round(yAxis.translate(
-                            point.y + yPad,
-                            0 as any,
-                            1 as any,
-                            0 as any,
-                            1 as any
-                        ) as any),
-                        -yAxis.len, 2 * yAxis.len
-                    ),
-                    pointPadding = pick(point.pointPadding, seriesPointPadding);
-
-                // Set plotX and plotY for use in K-D-Tree and more
-                point.plotX = point.clientX = (x1 + x2) / 2;
-                point.plotY = (y1 + y2) / 2;
-
-                point.shapeType = 'rect';
-                point.shapeArgs = {
-                    x: Math.min(x1, x2) + pointPadding,
-                    y: Math.min(y1, y2) + pointPadding,
-                    width: Math.max(Math.abs(x2 - x1) - pointPadding * 2, 0),
-                    height: Math.max(Math.abs(y2 - y1) - pointPadding * 2, 0)
+                var pointAttr,
+                    dif,
+                    hasImage =
+                        (point.marker && point.marker.symbol || shape || '')
+                            .match(/url/),
```

Comment:
Why we have two different checks for `hasImage`? In `markerAttribs` we have: 
> point.hasImage = (symbol && symbol.indexOf('url') === 0) || false;

Can we perhaps use `point.hasImage` here? And unify logic in `markerAttribs`?

Reply:
Agree. From now on, setting the `point.hasImage` just once within `series.translate()`, and only read that property in `markerAttribs()` function.